BusinessTampereTrafficMonitoring/traffic_lights/api_client.py

```diff
@@ -79,17 +96,19 @@ def store(self, events: List):
                     signal_group=signal_group,
                     t_start=_parse_date(t_start),
                     t_green=_parse_date(t_green),
-                    t_end=_parse_date(t_end)
-                )
+                    t_end=_parse_date(t_end))
                 db_conn.execute(stmt)
-            db_conn.commit()
+                db_conn.commit()
```

Comment:
This shouldn't have been changed. It is only necessary to do `db_conn.commit()` once, after the loop.

Reply:
But there is a catch if we do commit outside of the loop that only finalize the last insert changes?

Reply:
No, it commits all of them. It works like a database transaction.

Reply:
I don't think so. I tried a similar insertion in another project but the inserts were not reflecting except the last one. Have you tested it?

Reply:
I just quickly tested it just to be 100% sure and it works:

```python
import sqlalchemy
db = sqlalchemy.create_engine("sqlite:///testdb.db", future=True)
table = sqlalchemy.table("testings", sqlalchemy.column("v"))
with db.connect() as conn:
    for i in range(10):
        stmt = table.insert().values(v=str(i))
        conn.execute(stmt)
    conn.commit()
```

```
sqlite> select * from testings;
0
1
2
3
4
5
6
7
8
9
```